crates/runtime/src/component/keyvalue.rs

```diff
@@ -66,6 +66,88 @@ where
     }
 }
 
+#[async_trait]
+impl<H> batch::Host for Ctx<H>
+where
+    H: Handler,
+{
+    #[instrument(skip_all, fields(num_keys = keys.len()))]
+    async fn get_many(
+        &mut self,
+        bucket: Resource<store::Bucket>,
+        keys: Vec<String>,
+    ) -> anyhow::Result<Result<Vec<Option<(String, Vec<u8>)>>>> {
+        let bucket = self.table.get(&bucket).context("failed to get bucket")?;
+        // NOTE(thomastaylor312): I don't like allocating a new vec, but I need borrowed strings to
+        // have the right type
+        let keys = keys.iter().map(String::as_str).collect::<Vec<_>>();
+
+        match wrpc::wrpc::keyvalue::batch::get_many(
+            &self.handler,
+            Some(ReplacedInstanceTarget::KeyvalueBatch),
+            bucket,
+            &keys,
+        )
+        .await?
+        {
+            Ok(res) => Ok(Ok(res
+                .into_iter()
+                .map(|opt| opt.map(|(k, v)| (k.to_string(), Vec::from(v))))
```

Comment:
```suggestion
                .map(|opt| opt.map(|(k, v)| (k, Vec::from(v))))
```

isn't `k` a `String` already?